fluent.syntax/fluent/syntax/ast.py

```diff
@@ -286,29 +283,64 @@ class Expression(SyntaxNode):
     """An abstract base class for expressions."""
 
 
-class StringLiteral(Expression):
-    def __init__(self, raw, value, **kwargs):
-        super(StringLiteral, self).__init__(**kwargs)
-        self.raw = raw
-        self.value = value
-
-
-class NumberLiteral(Expression):
+class Literal(Expression):
+    """An abstract base class for literals."""
     def __init__(self, value, **kwargs):
-        super(NumberLiteral, self).__init__(**kwargs)
+        super(Literal, self).__init__(**kwargs)
         self.value = value
 
+    def parse(self):
+        return {'value': self.value}
+
+
+class StringLiteral(Literal):
+    def parse(self):
+        def from_escape_sequence(matchobj):
+            c, codepoint4, codepoint6 = matchobj.groups()
+            if c:
+                return c
+            codepoint = int(codepoint4 or codepoint6, 16)
+            if codepoint <= 0xD7FF or 0xE000 <= codepoint:
+                return six.unichr(codepoint)
+            # Escape sequences reresenting surrogate code points are
+            # well-formed but invalid in Fluent. Replace them with U+FFFD
+            # REPLACEMENT CHARACTER.
+            return '�'
+
+        value = re.sub(
+            r'\\(?:(\\|")|u([0-9a-fA-F]{4})|U([0-9a-fA-F]{6}))',
+            from_escape_sequence,
+            self.value
+        )
+        return {'value': value}
+
+
+class NumberLiteral(Literal):
+    def parse(self):
+        value = float(self.value)
+        decimal_position = self.value.find('.')
+        precision = 0
+        if decimal_position >= 0:
+            precision = len(self.value) - decimal_position - 1
+        return {
+            'value': value,
+            'precision': precision
+        }
+
 
 class MessageReference(Expression):
-    def __init__(self, id, **kwargs):
+    def __init__(self, id, attribute=None, **kwargs):
```

Comment:
In the js impl, the names here are still `attr`, but with actual keyword arguments in python, these need to match the properties for `traverse` to continue to work.